### PR TITLE
Expand on meaning and examples of Relationship Kinds

### LINKED ISSUE #3032: task: Docs Improvement - Expand on meaning and examples of Relationship Kinds
### Component

Documentation & Examples

### Task Description

From @ubaumann via discord:
> What do the different Relationship Kinds mean? I saw in the Reference what kinds I could use, but I could not find the meanings. It would be nice if the Schema References would have examples for the items.

### PR DESCRIPTION
**Pull request by BeArchiTek**
Fixes https://github.com/opsmill/infrahub/issues/3032

### CODE REVIEW

File: docs/docs/topics/schema.mdx [hunk 1]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+- `Component`: This relationship indicates that one entity is a part of another and typically appears in a separate tab. It represents a composition-like relationship where one node is a component of the current node.
+- `Parent`: This relationship defines a hierarchical link, with the parent entity often serving as a container or owner of another node. **Parent relationships are mandatory** and allow filtering in the UI, such as showing all children for a given parent.
+- `Group`: Defines a relationship where a node (inheriting from `CoreNode`) is a member or subscriber to a group (inheriting from `CoreGroup`). These relationships appear in the "Manage Groups" form.
+- `Profile`: A specialized relationship where a node is assigned a profile (inheriting from `CoreProfile`), visible during creation or updates through a "select profile" dropdown.
+
+To help you understand the relationship types better, here’s an example schema using a real-world model of **Car**, **Person**, **Engine**, and **CarGroup**.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: Engine seems to be missing in this example schema

File: docs/docs/topics/schema.mdx [hunk 2]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
 
 #### Relationship kinds
 
-- `Generic`: Default relationship without specific significance
-- `Attribute`: Relationship of type Attribute are represented in the detailed view and the list view
-- `Component`: Indicate a relationship with another node that is a component of the current node. Example: Interface is a component to a Device
-- `Parent`: Indicate a relationship with another node that is a parent to the current node. Example: Device is a parent to an Interface
-- `Group`: Indicate a relationship to a member or a subscriber of a group
+- `Generic`: A flexible relationship that has no specific functional significance. It is commonly used when an entity doesn't fit into more specialized categories like `Component` or `Parent`.
+- `Attribute`: A relationship where related entities' attributes appear directly in the detailed view and can influence list views. It's used for linking key information, like statuses or roles.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: `and can influence list views.` > and list view

File: docs/docs/topics/schema.mdx [hunk 3]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+- `Component`: This relationship indicates that one entity is a part of another and typically appears in a separate tab. It represents a composition-like relationship where one node is a component of the current node.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: and typically appears in a separate tab > appears in a separate tab

File: docs/docs/topics/schema.mdx [hunk 4]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+- `Parent`: This relationship defines a hierarchical link, with the parent entity often serving as a container or owner of another node. **Parent relationships are mandatory** and allow filtering in the UI, such as showing all children for a given parent.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: showing all children > showing all components
maybe we should highlight that component / parent relationships typically belong together

File: docs/docs/topics/schema.mdx [hunk 5]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+- `Group`: Defines a relationship where a node (inheriting from `CoreNode`) is a member or subscriber to a group (inheriting from `CoreGroup`). These relationships appear in the "Manage Groups" form.
+- `Profile`: A specialized relationship where a node is assigned a profile (inheriting from `CoreProfile`), visible during creation or updates through a "select profile" dropdown.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: Should we mention that users typically shouldn't use this, or even flag them for internal usage?

File: docs/docs/topics/schema.mdx [hunk 6]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
-- `Group`: Indicate a relationship to a member or a subscriber of a group
+- `Generic`: A flexible relationship that has no specific functional significance. It is commonly used when an entity doesn't fit into more specialized categories like `Component` or `Parent`.
+- `Attribute`: A relationship where related entities' attributes appear directly in the detailed view and can influence list views. It's used for linking key information, like statuses or roles.
+- `Component`: This relationship indicates that one entity is a part of another and typically appears in a separate tab. It represents a composition-like relationship where one node is a component of the current node.
+- `Parent`: This relationship defines a hierarchical link, with the parent entity often serving as a container or owner of another node. **Parent relationships are mandatory** and allow filtering in the UI, such as showing all children for a given parent.
+- `Group`: Defines a relationship where a node (inheriting from `CoreNode`) is a member or subscriber to a group (inheriting from `CoreGroup`). These relationships appear in the "Manage Groups" form.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx:98]: Should we mention that users typically shouldn't use this, or even flag them for internal usage?

File: docs/docs/topics/schema.mdx [hunk 7]
```diff
@@ -91,11 +91,110 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+        peer: AutoCar
+        kind: Generic
+        cardinality: many
+        optional: true
+```
+
+- A `CarGroup` has a generic relationship with `Car`, grouping multiple cars for fleet management.
+- A `Car` node might have an `owner` attribute linking it to a `Person`. This relationship will be visible in the car's detailed view.
+- An `Engine` is a component of a `Car`, meaning an engine is an essential part of the car. The engine will be displayed in a separate "Components" tab.
+- An `Interface` belongs to a `Device`, where the device is the parent of the interface. When selecting interfaces, you can filter them by their parent device.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: Interface?

**Review by wvandeun — CHANGES_REQUESTED** [review]
I think there's a few improvements to be done in the example and some of the explanations, see inline comments.

File: docs/docs/topics/schema.mdx (hunk 5 above)
Comment (BeArchiTek) [reply docs/docs/topics/schema.mdx]: I was thinking about this. I did try to apply a CoreProfile to a node but it seems to failed so gonna update those section too

File: docs/docs/topics/schema.mdx [hunk 8]
```diff
@@ -91,11 +91,125 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+- `Attribute`: A relationship where related entities' attributes appear directly in the detailed view and list views. It's used for linking key information, like statuses or roles.
+- `Component`: This relationship indicates that one entity is a part of another and appears in a separate tab. It represents a composition-like relationship where one node is a component of the current node.
+- `Parent`: This relationship defines a hierarchical link, with the parent entity often serving as a container or owner of another node. **Parent relationships are mandatory** and allow filtering in the UI, such as showing all components for a given parent.
+- `Group`: Defines a relationship where a node (inheriting from `CoreNode`) is a member or subscriber to a group (inheriting from `CoreGroup`). These relationships appear in the "Manage Groups" form.
+- `Profile`: A specialized relationship where a node is assigned a profile (inheriting from `CoreProfile`), visible during creation or updates through a "select profile" dropdown.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: A special reltionshhip

File: docs/docs/topics/schema.mdx [hunk 9]
```diff
@@ -91,11 +91,125 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
 #### Relationship kinds
 
-- `Generic`: Default relationship without specific significance
-- `Attribute`: Relationship of type Attribute are represented in the detailed view and the list view
-- `Component`: Indicate a relationship with another node that is a component of the current node. Example: Interface is a component to a Device
-- `Parent`: Indicate a relationship with another node that is a parent to the current node. Example: Device is a parent to an Interface
-- `Group`: Indicate a relationship to a member or a subscriber of a group
+- `Generic`: A flexible relationship that has no specific functional significance. It is commonly used when an entity doesn't fit into more specialized categories like `Component` or `Parent`.
+- `Attribute`: A relationship where related entities' attributes appear directly in the detailed view and list views. It's used for linking key information, like statuses or roles.
+- `Component`: This relationship indicates that one entity is a part of another and appears in a separate tab. It represents a composition-like relationship where one node is a component of the current node.
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: in a separate tab in the detailed view of a node in the UI

File: docs/docs/topics/schema.mdx [hunk 10]
```diff
@@ -91,11 +91,125 @@ The `kind` of a model is generated by concatenating the `namespace` and the `nam
...
+- `Parent`: This relationship defines a hierarchical link, with the parent entity often serving as a container or owner of another node. **Parent relationships are mandatory** and allow filtering in the UI, such as showing all components for a given parent.
+- `Group`: Defines a relationship where a node (inheriting from `CoreNode`) is a member or subscriber to a group (inheriting from `CoreGroup`). These relationships appear in the "Manage Groups" form.
+- `Profile`: A specialized relationship where a node is assigned a profile (inheriting from `CoreProfile`), visible during creation or updates through a "select profile" dropdown.
+
+:::info Complementary relationship
+
+**Component and Parent typically belong together**:
```
Comment (wvandeun) [inline docs/docs/topics/schema.mdx]: I would not make this line bold, IMO it renders a bit weird

**Review by wvandeun — CHANGES_REQUESTED** [review]
Few small things.

One bigger thing is that we say that profiles and groups are relationship kinds that a user should not use, but then we use the group relationship in the schema. Either we need to clarify this better or we should not add it to the example.

**Review by dgarros — COMMENTED** [review]
Would be good to change the target branch for this PR to `release-1.0` and include it in the release.

**Comment by BeArchiTek** [comment]
> Would be good to change the target branch for this PR to `release-1.0` and include it in the release.

I was planning to merge it and then do a cherry-pick to `release-1.0`

**Review by dgarros — APPROVED** [review]

**Pull request merged**